metaflow/plugins/argo/argo_workflows.py

```diff
@@ -1916,6 +1931,107 @@ def _slack_success_template(self):
             Http("POST").url(self.notify_slack_webhook_url).body(json.dumps(payload))
         )
 
+    def _heartbeat_daemon_template(self):
+        # Use any step that is known to exist, as we only care
+        # about being able to retrieve the code package for the utility scripts.
+        executable = self.environment.executable("_parameters")
+        run_id_template = "argo-{{workflow.name}}"
+        entrypoint = [executable, "-m metaflow.plugins.argo.heartbeat_daemon "]
+        heartbeat_cmds = "%s %s %s" % (
+            " ".join(entrypoint),
+            self.flow.name,
+            run_id_template,
+        )
+
+        # TODO: we do not really need MFLOG logging for the daemon at the moment, but might be good for the future.
+        # Consider if we can do without this setup.
+        # Configure log capture.
+        mflog_expr = export_mflog_env_vars(
```

Comment:
Yeah we can do without mflog for now. Given that this is an underlying implementation detail at the moment that shouldn’t concern the user.

Reply:
can't actually do much about this while still relying on environment internals for package download, as `mflog()` bash function setup is part of the `get_package_commands` which is littered with calls to `mflog`.

the env vars exported by `export_mflog_env_vars` are required for the logging, otherwise things break. I can either keep the existing setup, which actually sets up working env vars. Alternatively I can opt for completely dummy values. easy way forward is to not touch the setup which guarantees it works